server/src/main/java/org/opensearch/action/support/clustermanager/ClusterManagerNodeRequest.java

```diff
@@ -110,6 +114,11 @@ public final Request masterNodeTimeout(String timeout) {
         return clusterManagerNodeTimeout(timeout);
     }
 
+    public final Request setRemoteRequest(boolean remoteRequest) {
+        this.remoteRequest = remoteRequest;
```

Comment:
I am still unclear why do we need this flag? The action knows where it should be executed, the retry listener should just help run the same action after scheduled delay on same threadpool

Reply:
We need this distinguishion between requests coming to Master node as we have same code block (In `TransportClusterManagerNodeAction`) executing for both the case (local/remote master) and our retry logic is also on top of it only. 

Using this flag we will determine whether request is generated from local node or from remote node. If it is local node's request we need to perform the retries on this node. If it is remote node's request, we will not perform retries on this node and let remote node perform the retries.

If request is from remote data node, then data node will set remoteRequest flag in {@link MasterNodeRequest} and send request to master, using that on master node we can determine if the request was localRequest or remoteRequest.